src/ast/passes/codegen_llvm.cpp

```diff
@@ -3528,11 +3529,200 @@ void CodegenLLVM::createPrintNonMapCall(Call &call, int &id)
+
+    auto max_entries = bpftrace_.config_.get(ConfigKeyInt::max_map_keys);
+    auto map_type = libbpf::BPF_MAP_TYPE_UNSPEC;
+    if (val_type.IsCountTy() && key->second.args_.empty())
+    {
+      max_entries = 1;
+      map_type = libbpf::BPF_MAP_TYPE_PERCPU_ARRAY;
+    }
+    else if (bpftrace_.feature_->has_map_percpu_hash() &&
+             (val_type.IsHistTy() || val_type.IsLhistTy() ||
+              val_type.IsCountTy() || val_type.IsSumTy() ||
+              val_type.IsMinTy() || val_type.IsMaxTy() || val_type.IsAvgTy() ||
+              val_type.IsStatsTy()))
+    {
+      map_type = libbpf::BPF_MAP_TYPE_PERCPU_HASH;
+    }
+    else
+    {
+      map_type = libbpf::BPF_MAP_TYPE_HASH;
+    }
+
+    createMapDefinition(name, map_type, max_entries, key->second, val_type);
+  }
+
+  // bpftrace internal maps
+  for (const StackType &stack_type : resources.stackid_maps)
+  {
+    // bpf_get_stackid() is kind of broken by design.
+    //
+    // First, stacks can suffer from hash collisions and we can lose stacks. We
+    // receive collision errors in userspace b/c we do not set
+    // BPF_F_REUSE_STACKID. Note that we should **NEVER** set this flag b/c then
+    // we can silently receive unrelated stacks in userspace.
+    //
+    // Second, there is not a great way to reduce the frequency of collision
+    // (other than increasing map size) b/c multiple probes can be sharing the
+    // same stack (remember they are hashed). As a result, we cannot delete
+    // entries in this map b/c we don't know when all the sharers are gone.
+    //
+    // Fortunately, we haven't seen many bug reports about missing stacks (or
+    // maybe people don't care that much). The proper solution would be to use
+    // bpf_get_stack() to get the actual stack trace and pass it to userspace
+    // through the ring buffer.  However, there is additional complexity with
+    // this b/c we need to set up a percpu map to use as scratch space for
+    // bpf_get_stack() to write into. Using the stack directly wouldn't work
+    // well b/c it would take too much stack space.
+    //
+    // The temporary fix is to bump the map size to 128K. Any further bumps
+    // should warrant consideration of the previous paragraph.
+    createMapDefinition(stack_type.name(),
+                        libbpf::BPF_MAP_TYPE_STACK_TRACE,
+                        128 << 10,
+                        MapKey({ CreateInt32() }),
+                        CreateArray(stack_type.limit, CreateUInt64()));
+  }
+
+  if (resources.needs_join_map)
+  {
+    auto value_size = 8 + 8 + bpftrace_.join_argnum_ * bpftrace_.join_argsize_;
+    SizedType value_type = CreateArray(value_size, CreateInt8());
+    createMapDefinition(to_string(MapManager::Type::Join),
+                        libbpf::BPF_MAP_TYPE_PERCPU_ARRAY,
+                        1,
+                        MapKey({ CreateInt32() }),
+                        value_type);
+  }
+
+  if (resources.needs_elapsed_map)
+  {
```

Comment:
Are we following the kernel convention of removing the brackets for single lines after conditionals?

Reply:
I don't think that we have a rule for that. I generally omit the brackets if there's just a single line (in this case, there're multiple lines, although it's just one function call, so I'd leave the brackets).